library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -505,9 +505,10 @@ void PlatformBridgeFilter::FilterBase::onResume() {
     internal_buffer->drain(internal_buffer->length());
     internal_buffer->addBufferFragment(
         *Buffer::BridgeFragment::createBridgeFragment(*result.pending_data));
-    free(result.pending_data);
   } else if (result.pending_data) {
     addData(*result.pending_data);
+    // result.pending_data is copied either in the case above (to internal_buffer) or in the line
+    // above (via addData) or both. Regardless, it can now be freed.
     free(result.pending_data);
```

Comment:
I have also considered rewriting this code like:

```c++
  if (result.pending_data) {
    addData(*result.pending_data);
    
    if (internal_buffer) {
      internal_buffer->drain(internal_buffer->length());
      internal_buffer->addBufferFragment(
        *Buffer::BridgeFragment::createBridgeFragment(*result.pending_data));
    }

    free(result.pending_data);
  }
```

It might be easier to follow. But it does get rid of the explicit illegal state assertion.